dice-game/service/src/main.rs

```diff
@@ -0,0 +1,153 @@
+use map_vec::Map;
+use oasis_std::{Address, Context};
+use rand::Rng;
+
+#[derive(oasis_std::Service)]
+struct DiceGame {
+    max_score: u32,
+    num_players: u32,
+    scores: Map<Address, u32>,
+}
+
+type Result<T> = std::result::Result<T, String>;
```

Comment:
```suggestion
```
You can just use `Result<whatever, String>` inline. It's clearer that way when you have a non-crate defined error type.